### PR TITLE
Fix private forked repo. CI

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

We need the token `CIRCLE_TOKEN` to be used in the generated CircleCI config file.

### CODE REVIEW

File: .circleci/create_circleci_config.py [hunk 1]
```diff
@@ -371,9 +371,14 @@ def create_circleci_config(folder=None):
             **{j.job_name + "_test_list":{"type":"string", "default":''} for j in jobs},
             **{j.job_name + "_parallelism":{"type":"integer", "default":1} for j in jobs},
         },
-        "jobs" : {j.job_name: j.to_dict() for j in jobs},
-        "workflows": {"version": 2, "run_tests": {"jobs": [j.job_name for j in jobs]}}
+        "jobs" : {j.job_name: j.to_dict() for j in jobs}
     }
+    if "CIRCLE_TOKEN" in os.environ:
+        # For private forked repo. (e.g. new model addition)
+        config["workflows"] = {"version": 2, "run_tests": {"jobs": [{j.job_name: {"context": ["TRANSFORMERS_CONTEXT"]}} for j in jobs]}}
```
Comment (ydshieh) [inline .circleci/create_circleci_config.py:378]: The token is defined within a context `TRANSFORMERS_CONTEXT` (set in CircleCI organization setting)

File: .circleci/create_circleci_config.py [hunk 2]
```diff
@@ -371,9 +371,14 @@ def create_circleci_config(folder=None):
             **{j.job_name + "_test_list":{"type":"string", "default":''} for j in jobs},
             **{j.job_name + "_parallelism":{"type":"integer", "default":1} for j in jobs},
         },
-        "jobs" : {j.job_name: j.to_dict() for j in jobs},
-        "workflows": {"version": 2, "run_tests": {"jobs": [j.job_name for j in jobs]}}
+        "jobs" : {j.job_name: j.to_dict() for j in jobs}
     }
+    if "CIRCLE_TOKEN" in os.environ:
```
Comment (ydshieh) [inline .circleci/create_circleci_config.py:376]: If this is available, it means we are in a repo. that is not `transformers` (i.e. private forked repo.)

It is determined by (in `config.yml`)

```
    setup_and_quality_2:
        when:
            not:
                 equal: [<<pipeline.project.git_url>>, https://github.com/huggingface/transformers]
        jobs:
            ...
            - fetch_tests:
                # [reference] https://circleci.com/docs/contexts/
                context:
                    - TRANSFORMERS_CONTEXT
```

**Comment by ydshieh** [comment]
cc @molbap FYI

**Review by LysandreJik — APPROVED** [review]
Top thanks @ydshieh!

**Pull request merged**